DRIVERS-1571 Retry on different mongos when possible (#1450)
mongodb/specifications: source/server-selection/server-selection.rst

2. If the topology wire version is invalid, raise an error and log a
   `"Server selection failed" message`_.

3. Find suitable servers by topology type and operation type. If a list of

>    deprioritized servers is provided, and the topology is sharded cluster,

Reviewer: ```suggestion
   deprioritized servers is provided, and the topology is a sharded cluster,
```

   these servers should be selected only if there are no other suitable servers.
   The server selection algorithm MUST ignore the deprioritized servers if the

>    topology is not sharded cluster.

Reviewer: ```suggestion
   topology is not a sharded cluster.
```

2. If the topology wire version is invalid, raise an error and log a
   `"Server selection failed" message`_.

3. Find suitable servers by topology type and operation type. In the case of

>    sharded clusters, a list of de-prioritized servers may be provided;

Reviewer 2: ```suggestion
   sharded clusters, a list of deprioritized servers may be provided;
```

5. If the topology wire version is invalid, raise an error and log a
   `"Server selection failed" message`_.

6. Find suitable servers by topology type and operation type. In the case of

>    sharded clusters, a list of de-prioritized servers may be provided;

Reviewer 2: ```suggestion
   sharded clusters, a list of deprioritized servers may be provided;
```

5. If the topology wire version is invalid, raise an error and log a
   `"Server selection failed" message`_.

6. Find suitable servers by topology type and operation type. If a list of

>    deprioritized servers is provided, and the topology is sharded cluster,

Reviewer: ```suggestion
   deprioritized servers is provided, and the topology is a sharded cluster,
```

   these servers should be selected only if there are no other suitable servers.
   The server selection algorithm MUST ignore the deprioritized servers if the

>    topology is not sharded cluster.

Reviewer: ```suggestion
   topology is not a sharded cluster.
```
